# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/trunk/webrtc/modules/desktop_capture/mouse_cursor_monitor_linux.cc
# Commit: 3f0981f37c1c
# Full Hash: 3f0981f37c1c499ad52016e526239b390703ba2a
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2019-04-17 21:47:29
# Regressor Bug: 1496359
# File Overlap Count: 1
# Description:
#   Bug 1496359 - Add PipeWire support to desktop capture; r=ng
#   
#   This is an import of upstream commit 318da51f99f91e3de1192f29d7f1824958f9f13e.
#   
#   Depends on D27367
# ==============================================================================

diff -r ceed4e0b3373 -r 3f0981f37c1c media/webrtc/trunk/webrtc/modules/desktop_capture/mouse_cursor_monitor_linux.cc
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/media/webrtc/trunk/webrtc/modules/desktop_capture/mouse_cursor_monitor_linux.cc	Mon Apr 15 17:26:27 2019 +0000
@@ -0,0 +1,51 @@
+/*
+ *  Copyright 2018 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree. An additional intellectual property rights grant can be found
+ *  in the file PATENTS.  All contributing project authors may
+ *  be found in the AUTHORS file in the root of the source tree.
+ */
+
+#include "modules/desktop_capture/mouse_cursor_monitor.h"
+
+#if defined(USE_X11)
+#include "modules/desktop_capture/linux/mouse_cursor_monitor_x11.h"
+#endif  // defined(USE_X11)
+
+namespace webrtc {
+
+// static
+MouseCursorMonitor* MouseCursorMonitor::CreateForWindow(
+    const DesktopCaptureOptions& options,
+    WindowId window) {
+#if defined(USE_X11)
+  return MouseCursorMonitorX11::CreateForWindow(options, window);
+#else
+  return nullptr;
+#endif  // defined(USE_X11)
+}
+
+// static
+MouseCursorMonitor* MouseCursorMonitor::CreateForScreen(
+    const DesktopCaptureOptions& options,
+    ScreenId screen) {
+#if defined(USE_X11)
+  return MouseCursorMonitorX11::CreateForScreen(options, screen);
+#else
+  return nullptr;
+#endif  // defined(USE_X11)
+}
+
+// static
+std::unique_ptr<MouseCursorMonitor> MouseCursorMonitor::Create(
+    const DesktopCaptureOptions& options) {
+#if defined(USE_X11)
+  return MouseCursorMonitorX11::Create(options);
+#else
+  return nullptr;
+#endif  // defined(USE_X11)
+}
+
+}  // namespace webrtc